### PR TITLE
Add vsphere-csi chart from helm3-charts

### PR DESCRIPTION
**Pull request by aiyengar2**
Ports vsphere-csi post merging https://github.com/rancher/helm3-charts/pull/62

### CODE REVIEW

**Comment by aiyengar2** [comment]
#### Note for Reviewers

This PR will be entirely rebased after https://github.com/rancher/helm3-charts/pull/62 is merged, which will impact the first commit and every commit that results in a Git conflict after that.

Probably makes more sense to give this PR a run-through after that has been merged. Will re-request for your reviews at that time and leave a comment to indicate this is ready to be reviewed for a merge.

Since the impending PR will cover private registry support + bubbling up images to the values.yaml, I will need to modify all of the images to point to the mirrored- versions of them after that rebase has occurred.

File: packages/vsphere-csi/charts/Chart.yaml [hunk 1]
```diff
@@ -9,3 +9,11 @@ icon: https://charts.rancher.io/assets/logos/vsphere-csi.svg
 maintainers:
   - name: Rancher
     email: caleb@rancher.com
+keywords:
+- infrastructure
+annotations:
+   catalog.cattle.io/certified: rancher
+   catalog.cattle.io/namespace: kube-system
+   catalog.cattle.io/release-name: vsphere-csi
+   catalog.cattle.io/display-name: "vSphere Cloud Storage Interface"
```
Comment (aiyengar2) [inline packages/vsphere-csi/charts/Chart.yaml:18]: @deniseschannon does this Display Name work or should it be short (e.g. `vSphere CSI`)

**Comment by aiyengar2** [comment]
This PR is ready for review.

This has been split up into a system-charts style PR where the first commit is a base copy (+ adding the package.yaml) and the following commits make the changes needed to get stuff added for the feature charts.

Probably easier to review commit by commit.

File: packages/vsphere-csi/charts/Chart.yaml (hunk 1 above)
Comment (deniseschannon) [reply packages/vsphere-csi/charts/Chart.yaml:18]: vSphere Cloud Storage Interface (CSI)

File: packages/vsphere-csi/charts/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+name: vsphere-csi
+version: 2.1.0
+appVersion: 2.1.0
+description: vSphere CSI driver
+sources:
+  - https://github.com/kubernetes-sigs/vsphere-csi-driver
+icon: https://charts.rancher.io/assets/logos/vsphere-csi.svg
+maintainers:
+  - name: Rancher
+    email: caleb@rancher.com
```
Comment (deniseschannon) [inline packages/vsphere-csi/charts/Chart.yaml:11]: Should we be using SUSE email addresses and should this be Steven?

Comment (deniseschannon) [reply packages/vsphere-csi/charts/Chart.yaml:11]: We discussed this and we will make a generic mailing list in the future for this, but won't change it for this release. Lucky @cbron

File: packages/vsphere-csi/charts/Chart.yaml (hunk 1 above)
Comment (aiyengar2) [reply packages/vsphere-csi/charts/Chart.yaml]: Updated, will update CPI as well.

File: packages/vsphere-csi/charts/values.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+vCenter:
+  host: ""
+  port: 443
+  insecureFlag: true
```
Comment (cbron) [inline packages/vsphere-csi/charts/values.yaml:4]: Should this be "1" or true ? @PennyScissors

Comment (aiyengar2) [reply packages/vsphere-csi/charts/values.yaml]: Switching to 1

**Review by cbron — DISMISSED** [review]
lgtm

**Review by cbron — APPROVED** [review]
We should probably clean up commits in rancher/charts more in future. Fine for now.

**Pull request merged**